model/kubernetes/kubernetes_cluster.rb

```diff
@@ -40,6 +40,18 @@ def path
   def endpoint
     api_server_lb.hostname
   end
+
+  def kubeconfig
+    rbac_token = cp_vms.first.sshable.cmd("sudo kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system get secret k8s-access -o jsonpath='{.data.token}' | base64 -d")
```

Comment:
huh what is this doing with root exactly?

Reply:
probably reading admin.conf I see. I wonder if you can down-scope the amount of root going on here with process redirection, i.e. `<(sudo cat /etc/kubernetes/admin.conf)`

Reply:
also, don't bother necessarily if you have a lot of other root kubectl calls that cannot be repair, no point gold-plating in one place.

Reply:
We run kubeadm with sudo during node provisioning, not sure if that's easy to escape.

Kubectl is used only here (for CP). And later with the upgrades.

Changing didn't hurt so just did it.